src/util/png.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+Minetest
+Copyright (C) 2021 hecks
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "png.h"
+#include <string>
+#include <sstream>
+#include <zlib.h>
+#include "util/serialize.h"
+#include "serialization.h"
+#include "irrlichttypes.h"
+
+void writeChunk(std::ostringstream &target, const std::string &chunk_str)
+{
+	writeU32(target, chunk_str.size() - 4); // Write length minus the identifier
+	target << chunk_str;
+	writeU32(target, crc32(0,(const u8*)chunk_str.data(), chunk_str.size()));
+}
+
+std::string encodePNG(const u8 *data, u32 width, u32 height, s32 compression)
+{
+	auto file = std::ostringstream(std::ios::binary);
+	file << "\x89PNG\r\n\x1a\n";
```

Comment:
cannot we use directly libpng for this ? it's one of our dependencies

Reply:
The server does not link to any of our graphics deps